newrelic/core/trace_cache.py

```diff
@@ -390,6 +395,47 @@ def record_event_loop_wait(self, start_time, end_time):
             root.increment_child_count()
             root.add_child(node)
 
+    # MutableMapping methods
+
+    def items(self):
+        for wr in self._cache.valuerefs():
+            value = wr()  # Dereferenced value is potentially no longer live.
+            if (
+                value is not None
+            ):  # weakref is None means weakref has been garbage collected and is no longer live. Ignore.
+                yield wr.key, value  # wr.key is the original dict key
+
+    def keys(self):
+        for wr in self._cache.valuerefs():
+            yield wr.key  # wr.key is the original dict key
```

Comment:
Should we have an ignore weakref None here as well in case the keys are used in a loop to grab the value?
```
for k in trace_cache.keys():
    value = trace_cache[k]
```

Reply:
I considered that, but it's slower to dereference and essentially useless from a correctness standpoint. At any point while iterating one of those can be garbage collected and then cause a KeyError. So it's really not much safer and just dereferences twice.

We don't use trace_cache.keys() anywhere currently so probably best to just leave a comment about that fact and only retrieve with .get(key, None).